Formula/miranda.rb

```diff
@@ -0,0 +1,63 @@
+class Miranda < Formula
+  desc "Lazy, purely functional programming language"
+  homepage "http://miranda.org.uk/"
+  url "https://www.cs.kent.ac.uk/people/staff/dat/miranda/src/mira-2066-src.tgz"
+  version "2.066"
+  sha256 "521c281e8c2fde87a2cd7c41d9677daa09514debb71435efc08ff3a7c20016eb"
+
+  depends_on "byacc" => :build
+
+  def install
+    # Make tasks are required to run sequentially
+    ENV.deparallelize
+
+    # Makefile assumes that provided paths exist
+    bin.mkpath
+    lib.mkpath
+    man1.mkpath
+
+    miralib = "#{lib}/miralib"
+
+    args = %W[
+      CC=#{ENV.cc}
+      CFLAGS=-O
+      BIN=#{bin}
+      LIB=#{lib}
+      MAN=#{man1}
+    ]
+
+    inreplace "Makefile" do |s|
+      # Skip compilation of `stdenv.m` & provided examples
+      s.gsub! " exfiles", ""
+      # Do NOT touch `#{lib}/miralib/` permissions
+      s.gsub! "./protect", "/usr/bin/true"
+      s.gsub! "./unprotect", "/usr/bin/true"
+      # Do NOT change `#{lib}/miralib/` ownership
+      s.gsub! "./ugroot", "whoami"
+    end
+
+    system "make", "cleanup"
+    system "make", "install", *args
+
+    # Remove compiled files, if any
+    rm_f "#{miralib}/preludx"
+    rm Dir["#{miralib}/**/*.x"]
+  end
+
+  def post_install
+    miralib = "#{lib}/miralib"
+
+    # Compile `stdenv.m` & provided examples
+    system "#{bin}/mira", "-make", "-lib", miralib, "#{miralib}/**/*.m"
```

Comment:
Why do these need to be compiled here?

Reply:
Strictly speaking, examples do not need to be precompiled but miralib aka `stdenv.m` needs to be compiled AOT because it is implicitly imported in any script you are trying to execute. I've excluded compilation from an original makefile by removing [`exfiles` rule](https://github.com/rljacobson/Miranda/blob/49d6dcc/Makefile#L74-L75) due to the issue described in my previous comment: https://github.com/Homebrew/homebrew-core/pull/51003#issuecomment-593660181
Which is why I re-introduced it as part of the post-install script.